### ISSUE TITLE
Support adding whole folders as dependencies

### ISSUE THREAD

**Issue by rhys-vdw** [usabilityhub/rails-erb-loader#10]
I *think* there's a way to do this in WebPack. Use case

```js
/* rails-erb-loader-dependencies ../config/locales/ */
export const AVAILABLE_LOCALES = <%= I18n.available_locales.to_json %>

```

**Comment by jdlehman** [comment]
If not, we can probably just check if the dependencies are files are folders. For folders we can walk through and add the files.

Might also be interesting to add whole folders/globs by adding glob support via something like: `https://github.com/isaacs/node-glob`

**Comment by rhys-vdw** [comment]
> If not, we can probably just check if the dependencies are files are folders. For folders we can walk through and add the files.

Before I added in the `rails-erb-loader-dependency` comment support, I tried an approach like this where by default each file would be rebuilt whenever any file matching `app/**/*.rb` was modified. The overhead of searching for files and registering watchers was significant. Like ten times slower than just rebuilding the files. I was not using `node-glob` though, I was using some other library (I think maybe `node-find-files` - I've deleted the branch now regrettably).

If node `node-glob` is faster and avoid this issue then yes, it would be cool to support globs. But if it's slow then I wouldn't want to present an API in which it's easy to naively slow your build right down. (Speed is priority number 1!)

A shortcoming of this approach that is file creation events would be missed too.

Anyways, it seems [`watchpack`](https://github.com/webpack/watchpack) supports directories, but I am not yet sure how to actually register them via the loader API yet.
